ext/posix/sys/socket.c

```diff
@@ -188,8 +194,6 @@ sockaddr_from_lua(lua_State *L, int index, struct sockaddr_storage *sa, socklen_
 	luaL_checktype(L, index, LUA_TTABLE);
 	family = checkintfield(L, index, "family");
 
-	memset(sa, 0, sizeof *sa);
```

Comment:
Only UNIX address family reqlly requires memset.

Reply:
why do you say that? I believe it is sensible to always zero out all the fields before populating with address family specific values, if only to let clients avoid trying to make sense of whatever bits were set in the uninitialised memory from malloc.

I could be persuaded to find a more portable way to do that where there are float or double-typed fields in one of the address family specific structures that `sa` gets cast to/from...

Reply:
Because of that:

![image](https://user-images.githubusercontent.com/1157412/42138191-aadbdd9c-7d92-11e8-860e-354cdca52fc5.png)

Other address families strictly specify their fields. And we set all these fields, so memset is superfluous  and just waste CPU cycles (if not eliminated by a compiler).

Yes, I can remove this fix in order to merge PR, but it is correct fix actually.